core/src/types/list.rs

```diff
@@ -80,44 +83,81 @@ impl Stream for Lister {
             return Poll::Ready(None);
         }
 
-        if let Some(fut) = self.stating.as_mut() {
-            let (path, rp) = ready!(fut.poll_unpin(cx));
-
-            // Make sure we will not poll this future again.
-            self.stating = None;
-            let metadata = match rp {
-                Ok(rp) => rp.into_metadata(),
+        while self.task_queue.len() < self.task_queue.capacity() {
```

Comment:
It's better not to add a loop in our `poll_next`, using `if` is enough?

Reply:
Changed to `if`, but I introduce a new field `listing` as `match` arms must have compatible type. I don't know whether the current implementation meets your expectations?